src/app/shared/components/course-form/course-form.component.ts

```diff
@@ -55,30 +106,45 @@ export class CourseFormComponent {
     );
 
     this.allAuthors.push(author);
-
-    const index = this.courseAuthorsArray.controls.findIndex(
-      c => c.value.id === author.id
-    );
-    if (index >= 0) {
-      this.courseAuthorsArray.removeAt(index);
-    }
+    const index = this.courseAuthorsArray.controls.findIndex(c => {
+      return c.value === author.id;
+    });
+    this.courseAuthorsArray.removeAt(index);
   }
 
   createNewAuthor() {
     const authorName = this.newAuthorGroup.get('author')?.value;
 
     if (authorName && authorName.length >= 2) {
       const newAuthor = {
-        id: uuidv4(),
         name: authorName,
       };
-      console.log(newAuthor);
+      this.courseStoreService.createAuthor(newAuthor);
 
-      this.allAuthors.push(newAuthor);
+      this.allAuthors.push(<Author>newAuthor.name);
       this.newAuthorGroup.reset();
     }
   }
 
+  onClickCancel() {
+    this.router.navigate(['courses']);
+  }
+
+  onSubmit() {
+    if (!this.courseForm.valid) {
+      this.courseForm.markAllAsTouched();
+      return;
+    }
+    if (this.isEditMode) {
+      this.courseStoreService
+        .editCourse(this.courseId, this.courseForm.value)
+        .subscribe();
```

Comment:
Subscribing to what exactly here?

Reply:
Well, without it, it didn't really edit the course, which makes sense but i guess it was because i tried to do it with "tap" inside the courses-store.
Anyway, i switched tap to subscribe in the course store so it works know without this empty subscription